apps/dashboard/src/pages/integrations/components/update-integration-sidebar.tsx

```diff
@@ -0,0 +1,150 @@
+  onClose: () => void;
+};
+
+export function UpdateIntegrationSidebar({ isOpened, integrationId, onClose }: UpdateIntegrationSidebarProps) {
+  const { integrations } = useFetchIntegrations();
+  const integration = integrations?.find((i) => i._id === integrationId);
+  const provider = novuProviders?.find((p) => p.id === integration?.providerId);
+
+  const { deleteIntegration, isLoading: isDeleting } = useDeleteIntegration();
+  const { mutateAsync: updateIntegration, isPending: isUpdating } = useUpdateIntegration();
+  const { mutateAsync: setPrimaryIntegration, isPending: isSettingPrimary } = useSetPrimaryIntegration();
+  const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false);
+
+  const {
+    isPrimaryModalOpen,
+    setIsPrimaryModalOpen,
+    pendingData,
+    handleSubmitWithPrimaryCheck,
+    handlePrimaryConfirm,
+    existingPrimaryIntegration,
+    isChannelSupportPrimary,
+  } = useIntegrationPrimaryModal({
+    onSubmit,
+    integrations,
+    integration,
+    mode: 'update',
+  });
+
+  async function onSubmit(data: IntegrationFormData) {
+    if (!integration) return;
+
+    /**
+     * We don't want to check the integration if it's a demo integration
+     * Since we don't have credentials for it
+     */
+    if (integration?.providerId === 'novu-email' || integration?.providerId === 'novu-sms') {
+      data.check = false;
+    }
+
+    try {
+      await updateIntegration({
+        integrationId: integration._id,
+        data: {
+          name: data.name,
+          identifier: data.identifier,
+          active: data.active,
+          primary: data.primary,
+          credentials: data.credentials,
+          check: data.check,
+        },
+      });
+
+      if (data.primary && data.active && isChannelSupportPrimary) {
+        await setPrimaryIntegration({ integrationId: integration._id });
+      }
+
+      onClose();
+    } catch (error: any) {
+      handleIntegrationError(error, 'update');
+    }
+  }
+
+  const onDelete = async () => {
+    if (!integration) return;
+
+    try {
+      await deleteIntegration({ id: integration._id });
+      toast.success('Integration deleted successfully');
+      setIsDeleteDialogOpen(false);
+      onClose();
+    } catch (error: any) {
+      handleIntegrationError(error, 'delete');
+    }
+  };
+
+  if (!integration || !provider) return null;
+
+  return (
+    <>
+      <IntegrationSheet isOpened={isOpened} onClose={onClose} provider={provider} mode="update">
+        <div className="scrollbar-custom flex-1 overflow-y-auto">
+          <IntegrationConfiguration
+            isChannelSupportPrimary={isChannelSupportPrimary}
+            provider={provider}
+            integration={integration}
+            onSubmit={handleSubmitWithPrimaryCheck}
+            mode="update"
+          />
+        </div>
+
+        <div className="bg-background flex justify-between gap-2 border-t p-3">
+          {integration.channel !== ChannelTypeEnum.IN_APP && (
```

Comment:
Mostly because I want to promote it's usage more. So it's always appereaing just with a different "connected" state and CTA. What would be the user benefit in your opinion of adding it?